File: xamarinmvvmcross/PropertyCross.Core/Domain/Model/PropertyLocationsResult.cs
using System.Collections.Generic;
using Newtonsoft.Json.Linq;

namespace PropertyCross.Core.Domain.Model
{
  public class PropertyLocationsResult : PropertyDataSourceResult
  {
    public PropertyLocationsResult(JObject json)
    {
      Data = new List<Location>();

      JArray listings = (JArray)json["response"]["locations"];
      foreach (var listing in listings)
      {
        Data.Add(new Location(listing));
      }
    }

    public List<Location> Data { get; private set; }
  }
}

File: native/android/PropertyCross/app/src/main/java/com/propertycross/android/model/PropertyLocationsResult.java
package com.propertycross.android.model;

import java.util.ArrayList;
import java.util.List;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class PropertyLocationsResult extends PropertyDataSourceResult {

	private List<Location> data;

	public PropertyLocationsResult(JSONObject result) {
		data = new ArrayList<Location>();
		try {
			JSONObject response = result.getJSONObject("response");

			JSONArray location = response.getJSONArray("locations");
			for (int i = 0; i < location.length(); i++) {
				data.add(new Location(location.getJSONObject(i)));
			}
		}
		catch (JSONException e) {
			data.clear();
		}
	}

	public List<Location> getData() {
		return data;
	}
}